pkg/actuators/machine/machine_scope.go

```diff
@@ -69,8 +70,18 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 		credentialsSecretName = providerSpec.CredentialsSecret.Name
 	}
 
+	minimalPowerVSClient, err := params.powerVSMinimalClient(params.client)
+	if err != nil {
+		return nil, machineapierros.InvalidMachineConfiguration("failed creating minimalPowerVS client Error: %v", err)
+	}
+
+	serviceInstanceID, err := getServiceInstanceID(providerSpec.ServiceInstance, minimalPowerVSClient)
```

Comment:
we also need explore how often this is getting called? because listing all the service is really a costly affair, time to think about caching!